sklearn/cluster/hierarchical.py

```diff
@@ -552,12 +554,24 @@ def _hc_cut(n_clusters, children, n_leaves):
     n_leaves : int
         Number of leaves of the tree.
 
+    distance_threshold : int (optional)
+        The distance threshold to cluster at.
+        NOTE: You should set either n_cluster or distance_threshold, NOT both.
+
+    distances : ndarray, shape (n_nodes-1,) (optional)
+        Only used when distance_threshold is set.
+        distances[i] refers to the distance between children[i][0] and
+        children[i][1] when they are merged.
+
     Returns
     -------
     labels : array [n_samples]
         cluster labels for each point
 
     """
+    # cut the tree when distance threshold is reached
+    if distance_threshold:
```

Comment:
Maybe explain in the param docstring of `distance_threshold` that setting `distance_threshold` would implicitly mean setting `n_clusters` to this expression. 

what are your thoughts @agramfort, @jnothman